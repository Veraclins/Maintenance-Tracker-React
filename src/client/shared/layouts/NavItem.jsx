import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const NavItem = ({ closeNavbar, children, link }) => (
  <li className="menu-item">
    <NavLink
      onClick={closeNavbar}
      to={`${link}`}
    >
      {children}
    </NavLink>
    <span className="arrow" />
    <div className="clear" />
  </li>
);

NavItem.propTypes = {
  closeNavbar: PropTypes.func.isRequired,
  children: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
};


export default NavItem;
