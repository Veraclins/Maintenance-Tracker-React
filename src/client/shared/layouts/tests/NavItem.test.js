import React from 'react';
import { shallow } from 'enzyme';
import { NavLink } from 'react-router-dom';

import NavItem from '../NavItem';

const mockFunction = jest.fn();

describe('Test the NavItem component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <NavItem
        link="login"
        closeNavbar={mockFunction}
      >
        Login
      </NavItem>,
    );
    expect(wrapper.find('li')).toHaveLength(1);
    expect(wrapper.find(NavLink)).toHaveLength(1);
    expect(wrapper.find('div.clear')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
