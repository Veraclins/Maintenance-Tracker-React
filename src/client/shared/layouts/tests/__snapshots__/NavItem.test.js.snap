// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Header component renders without crashing 1`] = `
<li
  className="menu-item"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    onClick={[MockFunction]}
    to="login"
  >
    Login
  </NavLink>
  <span
    className="arrow"
  />
  <div
    className="clear"
  />
</li>
`;
